### PR TITLE
Added keycloak OpenIdConnect/OAuth2.0 Server integration docs for flyte

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

### CODE REVIEW

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 1]
```diff
@@ -102,7 +102,12 @@ Flyte supports connecting with external OIdC providers. Here are some examples f
 .. tabbed:: KeyCloak
 
     `KeyCloak <https://www.keycloak.org/>`__ is an open source solution for authentication, it supports both OpenID Connect and OAuth2 protocols (among others).
-    KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for Flyte.
+    KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for Flyte.Here we configure to use it for OpenID Connect.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:105]:
```suggestion
    KeyCloak can be configured as both the OpenID Connect and OAuth2 Authorization Server provider for Flyte. Here we configure to use it for OpenID Connect.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 2]
```diff
@@ -102,7 +102,12 @@ Flyte supports connecting with external OIdC providers. Here are some examples f
...
+
+    1. If you don't have a Keycloak installation then you can use `this <https://www.amazonaws.cn/en/solutions/keycloak-on-aws/>`__ which provides quick way to deploy Keycloak cluster on AWS.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:107]:
```suggestion
    1. If you don't have a Keycloak installation then you can use `this <https://www.amazonaws.cn/en/solutions/keycloak-on-aws/>`__ which provides a quick way to deploy Keycloak cluster on AWS.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 3]
```diff
@@ -102,7 +102,12 @@ Flyte supports connecting with external OIdC providers. Here are some examples f
...
+    2. Create a realm in keycloak installation using its `admin console <https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/realms/create.html>`__
+    3. Create an OIDC client with client secret and note them down. Use the following `instructions <https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/clients/client-oidc.html>`__
+    4. Add Login redirect URIs (e.g. http://localhost:30081/callback for sandbox or ``https://<your deployment url>/callback``)
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:110]:
```suggestion
    4. Add Login redirect URIs (e.g, http://localhost:30081/callback for sandbox or ``https://<your deployment url>/callback``)
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 4]
```diff
@@ -157,7 +163,7 @@ Apply Configuration
           # 4. Update with public domain name (for non-sandbox deployments)
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:163]:
```suggestion
          # 4. Update with a public domain name (for non-sandbox deployments).
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 5]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
 
 .. tabbed:: KeyCloak
 
-   `KeyCloak <https://www.keycloak.org/>`__ is an open source solution for authentication, it supports both OpenID Connect and OAuth2 protocols (among others).
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:208]: Is this "KeyCloak" or "Keycloak"?

File: rsts/deployment/cluster_config/auth_setup.rst (hunk 2 above)
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:107]: Is this "KeyCloak" or "Keycloak"?

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 6]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
-   KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for flyte.
+    `KeyCloak <https://www.keycloak.org/>`__ is an open source solution for authentication, it supports both OpenID Connect and OAuth2 protocols (among others).
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:214]:
```suggestion
    `KeyCloak <https://www.keycloak.org/>`__ is an open source solution for authentication. It supports both OpenID Connect and OAuth2 protocols (among others).
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 7]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
+    KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for Flyte.Here we use it as OAuth2 Authorization Server.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:215]: Is this "KeyCloak" or "Keycloak"?

Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:215]:
```suggestion
    KeyCloak can be configured as both the OpenID Connect and OAuth2 Authorization Server provider for Flyte. Here we use it as OAuth2 Authorization Server.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 8]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
+
+    1. If you don't have a Keycloak installation then you can use `this <https://www.amazonaws.cn/en/solutions/keycloak-on-aws/>`__ which provides quick way to deploy Keycloak cluster on AWS.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:217]:
```suggestion
    1. If you don't have a Keycloak installation, you can use `this <https://www.amazonaws.cn/en/solutions/keycloak-on-aws/>`__ which provides a quick way to deploy Keycloak cluster on AWS.
```

Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:217]: "KeyCloak" or "Keycloak"?

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 9]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
+    2. Create a realm in keycloak installation using its `admin console <https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/realms/create.html>`__
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:218]: "KeyCloak" or "Keycloak"?

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 10]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
+    3. Under `Client Scopes`, click `Add Create` inside the admin console
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:219]:
```suggestion
    3. Under `Client Scopes`, click `Add Create` inside the admin console.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 11]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
+    4. Create 2 clients (for flytectl and flytepropeller) to enable these clients to communicate with the service.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:220]:
```suggestion
    4. Create 2 clients (for Flytectl and Flytepropeller) to enable these clients to communicate with the service.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 12]
```diff
@@ -205,8 +211,15 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
...
+       * Flytectl should be created with `Access Type Public` and standard flow enabled.
+       * FlytePropeller should be created as an `Access Type Confidential`, standard flow enabled and note the client ID and client Secrets provided.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:222]:
```suggestion
       * FlytePropeller should be created as an `Access Type Confidential`, standard flow enabled, and note the client ID and client Secrets provided.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 13]
```diff
@@ -225,13 +238,16 @@ Apply Configuration
                 # 2. Optional: Set external auth server baseUrl if different from OpenId baseUrl.
                 externalAuthServer:
                     baseUrl: https://dev-14186422.okta.com/oauth2/auskngnn7uBViQq6b5d6
+                    #baseUrl: https://<keycloak-url>/auth/realms/<keycloak-realm> # Uncomment for keycloak
+                    #metadataUrl: .well-known/openid-configuration #Uncomment for keycloak
+
             thirdPartyConfig:
                 flyteClient:
-                    # 3. Replace with a new Native Client ID provisioned in the custom authorization server
+                    # 3. Replace with a new Native/Public Client ID provisioned in the custom authorization server
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:246]: Making it consistent with the above two points ('.' at the end).
```suggestion
                    # 3. Replace with a new Native/Public Client ID provisioned in the custom authorization server.
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 14]
```diff
@@ -225,13 +238,16 @@ Apply Configuration
...
                     clientId: flytectl
 
                     # This should not change
-                    redirectUri: https://localhost:53593/callback
+                    redirectUri: http://localhost:53593/callback
 
                     # 4. "all" is a required scope and must be configured in the custom authorization server
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:252]:
```suggestion
                    # 4. "all" is a required scope and must be configured in the custom authorization server.
```

File: rsts/deployment/cluster_config/auth_setup.rst (hunk 5 above)
Comment (pmahindrakar-oss) [reply rsts/deployment/cluster_config/auth_setup.rst:208]: Its Keycloak. fixed it in all places

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 15]
```diff
@@ -99,10 +99,15 @@ Flyte supports connecting with external OIdC providers. Here are some examples f
...
-    KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for Flyte.
+    `Keycloak <https://www.keycloak.org/>`__ is an open source solution for authentication.It supports both OpenID Connect and OAuth2 protocols (among others).
+    Keycloak can be configured as both the OpenID Connect and OAuth2 Authorization Server provider for Flyte. Here we configure to use it for OpenID Connect.
+
+    1. If you don't have a Keycloak installation, you can use `this <https://www.amazonaws.cn/en/solutions/keycloak-on-aws/>`__ which provides a quick way to deploy Keycloak cluster on AWS.
+    2. Create a realm in keycloak installation using its `admin console <https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/realms/create.html>`__
+    3. Create an OIDC client with client secret and note them down. Use the following `instructions <https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/clients/client-oidc.html>`__
+    4. Add Login redirect URIs (e.g, http://localhost:30081/callback for sandbox or ``https://<your deployment url>/callback``)
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:110]:
```suggestion
    4. Add Login redirect URIs (e.g, http://localhost:30081/callback for sandbox or ``https://<your deployment url>/callback``).
```

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 16]
```diff
@@ -199,14 +205,21 @@ To set up an external OAuth2 Authorization Server, please follow the instruction
    1. Under security -> API, click `Add Authorization Server`. Set the audience to the public URL of FlyteAdmin (e.g. https://flyte.mycompany.io/).
    2. Under `Access Policies`, click `Add New Access Policy` and walk through the wizard to allow access to the authorization server.
    3. Under `Scopes`, click `Add Scope`. Set the name to `all` (required) and check `Require user consent for this scope` (recommended).
-   4. Create 2 apps (for flytectl and flytepropeller) to enable these clients to communicate with the service.
+   4. Create 2 apps (for Flytectl and Flytepropeller) to enable these clients to communicate with the service.
       Flytectl should be created as a `native client`.
       FlytePropeller should be created as an `OAuth Service` and note the client ID and client Secrets provided.
```
Comment (AlekhyaSasi) [inline rsts/deployment/cluster_config/auth_setup.rst:210]:
```suggestion
      Flytepropeller should be created as an `OAuth Service` and note the client ID and client Secrets provided.
```

**Review by AlekhyaSasi — APPROVED** [review]

**Comment by pmahindrakar-oss** [comment]
Thanks @AlekhyaSasi for the review and the fixes

**Pull request merged**